gapis/api/vulkan/api/buffer.api

```diff
@@ -299,11 +311,21 @@ sub void BindBufferMemory2(
         for i in (0 .. numPNext) {
           sType := as!const VkStructureType*(next.Ptr)[0:1][0]
           _ = sType
-          // TODO: handle extensions for VkBindBufferMemoryInfo
+          switch sType {
+            case VK_STRUCTURE_TYPE_BIND_BUFFER_MEMORY_DEVICE_GROUP_INFO: {
+              ext := as!VkBindBufferMemoryDeviceGroupInfo*(next.Ptr)[0:1][0]
+              dg.Binding = new!DeviceGroupBinding()
+              indices := ext.pDeviceIndices[0:ext.deviceIndexCount]
+              for j in (0 .. ext.deviceIndexCount) {
+                dg.Binding.Bindings[j] = indices[j]
+              }
+            }
+            default: {}
+          }
           next.Ptr = as!VulkanStructHeader*(next.Ptr)[0:1][0].PNext
         }
       }
-      BindBufferMemory(info.buffer, info.memory, info.memoryOffset)
+      BindBufferMemory(info.buffer, info.memory, info.memoryOffset, dg.Binding)
```

Comment:
another learning question, why we are calling `BindBufferMemory` inside `BindBufferMemory2`?

Reply:
So we are calling BindBufferMemory from both `vkBindBufferMemory && vkBindBufferMemory2` a common function so we dont have to duplicate the logic.